### PR TITLE
Options rough draft

### PR DESCRIPTION
**Pull request by MattSchiller**
## Description

Just a demonstration of the type change that I think the generated method should be using.

I wish I had more time, but I really don't 😭 .  I hope this helps!

@cyberhck

### CODE REVIEW

File: packages/plugins/typescript/graphql-apollo/src/visitor.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ export class GraphQLApolloVisitor extends ClientSideBaseVisitor<
     const typeImport = this.config.useTypeImports ? 'import type' : 'import';
 
     this._additionalImports.push(`${typeImport} { ApolloClient } from '@apollo/client';`);
+    this._additionalImports.push(`${typeImport} * as Apollo from '@apollo/client';`);
```
Comment (MattSchiller) [inline packages/plugins/typescript/graphql-apollo/src/visitor.ts:35]: I'm not convinced this is the best way to update the imports, but... maybe it's good enough? 😁

Comment (cyberhck) [reply packages/plugins/typescript/graphql-apollo/src/visitor.ts:35]: actually let's not do that, I'll change it, the reason is it'll import everything and unless you have treeshaking (which is already tricky), you'll get everything, which we don't want.

**Review by cyberhck — COMMENTED** [review]
let me cherry pick this I hope you don't mind, I'll get code done before Monday :slightly_smiling_face:

File: packages/plugins/typescript/graphql-apollo/src/visitor.ts [hunk 2]
```diff
@@ -69,23 +70,26 @@ export class GraphQLApolloVisitor extends ClientSideBaseVisitor<
...
-      }>({variables, query: ${documentVariableName}})
+    const sdkOperations = this._operationsToInclude.map(
+      ({ node, documentVariableName, operationType, operationResultType, operationVariablesTypes }) => {
+        const optionalVariables =
+          !node.variableDefinitions ||
+          node.variableDefinitions.length === 0 ||
+          node.variableDefinitions.every(v => v.type.kind !== Kind.NON_NULL_TYPE || v.defaultValue);
+
+        const operationName = node.name.value;
+        const operationGenerics = `${operationResultType}, ${operationVariablesTypes}`;
```
Comment (cyberhck) [inline packages/plugins/typescript/graphql-apollo/src/visitor.ts:81]: @MattSchiller I don't know what's going on with apollo, but types don't make sense.

Have a look :smile: 
For mutations:
```typescript
mutate<T = any, TVariables = OperationVariables>(options: MutationOptions<T, TVariables>): Promise<FetchResult<T>>;
```
and for queries:
```
query<T = any, TVariables = OperationVariables>(options: QueryOptions<TVariables, T>): Promise<ApolloQueryResult<T>>;
```
now I've no idea why mutationOptions and queryOptions are flipped

The generator isn't going to look pretty, but I think I can make it work.

Comment (cyberhck) [reply packages/plugins/typescript/graphql-apollo/src/visitor.ts:81]: https://github.com/apollographql/apollo-client/issues/8537 created an issue :smile:

**Pull request closed without merging**